### PR TITLE
Migrate remaining turrets to gun_actor

### PR DESCRIPTION
**Pull request by mugling**
The `ammo_` accessors are now sufficiently complete that we can have generic turrets of the form:

```
    "starting_ammo": { "9mm": 100 },
    "special_attacks": [ {
        "type": "gun",
        "cooldown": 1,
        "gun_type": "hk_mp5"
    } ]
```

This new implementation supports both guns with magazines or those using no ammo/UPS.
- Moves all defaults out of `gun_actor` implementation to header
- Expand and clarify documentation for class members
- Provide default attack messages (aaf1c5d, 08f2a1c)
- Adds support for guns that don't use ammo (bba885b)
- Add support for turrets that require sunlight (9ad3724)
- Drop `mattack::mattack::laser` and `mattack::rifle_tur` (47e1114, ae5eaf0, c0a073e)
- Drop special case broken cerberus laser (ed3274e) causing `debugmsg` in `--dump-stats` (#15765)

### CODE REVIEW

**Comment by Mshock777** [comment]
Maybe for clarity rename `military` to `military_bot`?

**Comment by mugling** [comment]
> Maybe for clarity rename military to military_bot?

The aim is to be consistent with the `default_faction` field

File: src/player.h [hunk 1]
```diff
@@ -1133,8 +1133,8 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
         int tank_plut, reactor_plut, slow_rad;
         int oxygen;
         int stamina;
-        int recoil;
-        int driving_recoil;
+        int recoil = 0;
+        int driving_recoil = 0;
```
Comment (Coolthulhu) [inline src/player.h]: Why not in constructor? Everything else is initialized there.

Comment (mugling) [inline src/player.h]: Yes, fair enough it's out-of-place compared to the other members.

File: src/mattack_actors.cpp [hunk 2]
```diff
@@ -384,26 +387,28 @@ void gun_actor::shoot( monster &z, Creature &target ) const
...
 
     const auto distance = rl_dist( z.pos(), target.pos() );
-    int burst_size = std::min( burst_limit, tmp.weapon.burst_size() );
-    if( distance > range_no_burst ) {
-        burst_size = 1;
+
+    const int burst = std::min( distance <= range_no_burst ? burst_limit : 1, tmp.weapon.burst_size() );
+
+    if( g->u.sees( z ) ) {
+        add_msg( m_warning, _( description.c_str() ), z.name().c_str(), tmp.weapon.gun_type().c_str() );
```
Comment (BevapDin) [inline src/mattack_actors.cpp]: The result of `item::gun_type` needs to be translated when you're printing it. See implementation of that function (it returns an untranslated string) and its usage in `item::gunmod_compatible` (where it is translated for printing).

**Pull request merged**